lib/health.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+const hasMatchingMethod = (option, reqMethod) => {
+  return option.methods.some(method => {
+    return method.toLowerCase() === reqMethod.toLowerCase();
+  });
+};
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        // Test the method property against the current req method
+        if (option.methods && (hasMatchingMethod(option, req.method) !== true)) {
+          return memo;
+        }
+        // memoize the endpoints
+        memo.push(request(option.url).promise());
+        return memo;
+      }, []);
+
+      // Process all pending requests
+      return Promise.all(requests)
+        .then(() => next())
+        .catch(error => {
+          // Assign a code and let
+          // the error middleware handle it
+          error = new Error(error);
+          error.code = 'UNHEALTHY';
+          next(error);
```

Comment:
To serve a template that appropriately informs the user about the issue we'll need to update the error middleware and `hof-template-partials`.
I would suggest that we take the opportunity to more closely align the middleware templates with the middleware, and make an amendment to `hof` to read templates from the middleware as well as the partials module